src/ship_cmd.cpp

```diff
@@ -442,11 +456,38 @@ static Track ChooseShipTrack(Ship *v, TileIndex tile, DiagDirection enterdir, Tr
 
 	bool path_found = true;
 	Track track;
-	switch (_settings_game.pf.pathfinder_for_ships) {
-		case VPF_OPF: track = OPFShipChooseTrack(v, tile, enterdir, tracks, path_found); break;
-		case VPF_NPF: track = NPFShipChooseTrack(v, tile, enterdir, tracks, path_found); break;
-		case VPF_YAPF: track = YapfShipChooseTrack(v, tile, enterdir, tracks, path_found); break;
-		default: NOT_REACHED();
+
+	if (v->dest_tile == 0 || DistanceManhattan(tile, v->dest_tile) > 130) {
```

Comment:
Is 130 a reasonable distance? Seems a bit short really (depending on the map of course)

Reply:
It is the same distance that the UI requires when creating orders, so seemed a reasonable value to use.

Reply:
Made a const, and also added a little leeway (buoys don't have to be reached exactly which could increase the distance.)